scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java

```diff
@@ -274,7 +274,7 @@ private int selectAndStartTasks(Policy currentPolicy, Map<JobId, JobDescriptor>
                 LinkedList<EligibleTaskDescriptor> tasksToSchedule = new LinkedList<>();
                 int neededResourcesNumber = 0;
 
-                while (taskRetrievedFromPolicy.size() > 0 && neededResourcesNumber == 0) {
+                while (taskRetrievedFromPolicy.isEmpty() && neededResourcesNumber == 0) {
```

Comment:
shouldn't be while (!taskRetrievedFromPolicy.isEmpty() && ....     ??

Reply:
woops, this was while I was playing with SonarLit,,, I thought I reverted it,,,

Reply:
I agree with marco ! Good find